dotnet-efcore/EfCoreCodeFirstMigrationsInProd/Tests/EfCoreCodeFirstMigrationsInProdTests/StartupFactoryTests.cs

```diff
@@ -0,0 +1,50 @@
+using EfCoreCodeFirstMigrationsInProd.Startup;
+
+namespace EfCoreCodeFirstMigrationsInProdTests;
+
+public class StartupFactoryTests
+{
+    [Test]
+    public void GivenStartupFactory_WhenMigrationArgIsPresent_ThenReturnMigrationStartup()
+    {
+        var args = new[] {"--migrate"};
+        var factory = new StartupFactory();
+        
+        var startup = factory.GetStartup(args);
+        
+        Assert.That(startup, Is.InstanceOf<MigrationStartup>());
+    }
+    
+    [Test]
+    public void GivenStartupFactory_WhenNoArgIsPresent_ThenReturnWebApiStartup()
+    {
+        var args = Array.Empty<string>();
+        var factory = new StartupFactory();
+        
+        var startup = factory.GetStartup(args);
+        
+        Assert.That(startup, Is.InstanceOf<WebApiStartup>());
+    }
+    
+    [Test]
+    public void GivenStartupFactory_WhenNoMigrationArgIsPresent_ThenReturnWebApiStartup()
+    {
+        var args = new[] {"arg1", "--arg2", "migrate", "--migration"};
```

Comment:
Would you mind just renaming the last array item here to something else? Since the test has "NoMigration" in the name, but the last item in the array is `--migration`, it might be a bit confusing what the test is doing at first glance.

Could just remove the last item too, and rename the test `GivenStartupFactory_WhenMultipleArgsArePresentWithNoMigrationArg_ThenReturnWebApiStartup`, to contrast it with the test below this one.

Reply:
Last item removed, and test renamed